src/instructions/b.rs

```diff
@@ -0,0 +1,352 @@
+                        .0
+                    })
+                }
+                0b_10 => {
+                    let inst_opt = match funct3_value {
+                        0b_001 => Some(insts::OP_FSL),
+                        0b_101 => Some(insts::OP_FSR),
+                        _ => None,
+                    };
+                    inst_opt.map(|inst| {
+                        R4type::new(
+                            inst,
+                            rd(instruction_bits),
+                            rs1(instruction_bits),
+                            rs2(instruction_bits),
+                            rs3(instruction_bits),
+                        )
+                        .0
+                    })
+                }
+                0b_11 => {
+                    let inst_opt = match funct3_value {
+                        0b_001 => Some(insts::OP_CMIX),
+                        0b_101 => Some(insts::OP_CMOV),
+                        _ => None,
+                    };
+                    inst_opt.map(|inst| {
+                        R4type::new(
+                            inst,
+                            rd(instruction_bits),
+                            rs1(instruction_bits),
+                            rs2(instruction_bits),
+                            rs3(instruction_bits),
+                        )
+                        .0
+                    })
+                }
+                _ => None,
+            }
+        }
+        0b_0010011 => {
+            let top12_value = instruction_bits >> 20;
+            let top7_value = top12_value >> 5;
+            if top7_value == 0b_0110000 {
+                let rs2_value = rs2(instruction_bits);
+                let inst_opt = match (funct3_value, rs2_value) {
+                    (0b_001, 0b_00000) => Some(insts::OP_CLZ),
+                    (0b_001, 0b_00001) => Some(insts::OP_CTZ),
+                    (0b_001, 0b_00010) => Some(insts::OP_PCNT),
+                    (0b_001, 0b_00011) => Some(insts::OP_BMATFLIP),
+                    (0b_001, 0b_00100) => Some(insts::OP_SEXTB),
+                    (0b_001, 0b_00101) => Some(insts::OP_SEXTH),
+                    (0b_001, 0b_10000) => Some(insts::OP_CRC32B),
+                    (0b_001, 0b_10001) => Some(insts::OP_CRC32H),
+                    (0b_001, 0b_10010) => Some(insts::OP_CRC32W),
+                    (0b_001, 0b_10011) => Some(insts::OP_CRC32D),
+                    (0b_001, 0b_11000) => Some(insts::OP_CRC32CB),
+                    (0b_001, 0b_11001) => Some(insts::OP_CRC32CH),
+                    (0b_001, 0b_11010) => Some(insts::OP_CRC32CW),
+                    (0b_001, 0b_11011) => Some(insts::OP_CRC32CD),
+                    _ => None,
+                };
+                if inst_opt.is_some() {
+                    return inst_opt.map(|inst| {
+                        Rtype::new(inst, rd(instruction_bits), rs1(instruction_bits), rs2_value).0
+                    });
+                }
```

Comment:
avoids duplicate pattern matching
```suggestion
                if let Some(inst) = inst_opt {
                    return Some(
                        Rtype::new(inst, rd(instruction_bits), rs1(instruction_bits), rs2_value).0,
                    );
                }
```